app/models/user.rb

```diff
@@ -61,4 +61,96 @@ def generate_token(column)
       break unless User.exists?(column => self[column])
     end
   end
+
+  # @return [TrueClass, FalseClass]
+  def uploaded_before? = !user_resources.all.empty?
```

Comment:
Stelle dir vor, du hast hier bei `user_resources` mehrere tausende Files. Sagen wir so ein User wie ich, hat sogar Millionen von Files!

Du lädst alle diese Files mit `.all`, um dann in Ruby zu fragen, ob welche geladen sind oder nicht?! Danach verwirfst du alle diese Objekte im Speicher. Das ist extrem ineffizient und wird schnell dazu führen, dass dein Server nicht mehr arbeiten will. :)

Schaue dir hingegen diese Methode von Active Record an, die 0 Objekte erzeugt und dadurch viel schneller und effizienter arbeitet:
https://apidock.com/rails/ActiveRecord/FinderMethods/exists%3F

Löse die Probleme nicht neu, benutze bestehende Lösungen! Dafür hast du dieses Framework.